### ISSUE TITLE
Remote server close on Fetch ALL

### ISSUE THREAD

**Issue by jvilhuber** [jstedfast/MailKit#183]
Just ran into something curious (I'm an IMAP newbie so perhaps this is expected?). First some info: I'm running without SSL to port 143 in the iOS simulator for now until I get my bearings. Code compiled under xamarin (let me know if you need any version information). My code is based mostly on the example code you provided, so nothing terribly out of the ordinary.
```
var folder = client.Inbox;
folder.Open(FolderAccess.ReadOnly);
var query = SearchQuery.DoesNotHaveFlags (MessageFlags.Deleted);
var uids = folder.Search (query);
var messageSummaries = folder.Fetch (uids, MessageSummaryItems.All);
```
This results in the following IMAP exchange:
```
C: A00000007 UID SEARCH RETURN () UNDELETED
S: * ESEARCH (TAG "A00000007") UID ALL 8622:8623,8630:8639
S: A00000007 OK SEARCH completed (Success)
C: A00000008 UID FETCH 8622:8623,8630:8639 ALL
```
But it throws an exception from the fetch, indicating that the IMAP server has closed the connection. I have confirmed this (seemingly) with wireshark (I'm using the ios simulator, so sniffing is via lo0, but I'm reasonably sure it's an accurate trace).

If I change the MessageSummaryItems to Envelope (which changes the IMAP query to `A00000008 UID FETCH 8622:8623,8630:8639 ENVELOPE`), then the fetch succeeds.

Is this expected?

**Comment by jstedfast** [comment]
It sounds like a bug in your IMAP server :-(

**Comment by jstedfast** [comment]
Possible workaround: `folder.Fetch (uids, MessageSummaryItems.All | MessageSummaryItems.UniqueId);`

**Comment by jvilhuber** [comment]
The IMAP server in question is imap.gmail.com :) Though not impossible that they have a bug, it does seem a bit unlikely (then again...). I can work around it, for sure.

**Comment by jstedfast** [comment]
I can prove that it's a server bug :-)

From rfc3501:

```
fetch           = "FETCH" SP sequence-set SP ("ALL" / "FULL" / "FAST" /
                  fetch-att / "(" fetch-att *(SP fetch-att) ")")
uid             = "UID" SP (copy / fetch / search / store)
                    ; Unique identifiers used instead of message
                    ; sequence numbers
```

Does MailKit's UID FETCH command not follow that syntax? :-)

**Comment by jvilhuber** [comment]
Thanks!

**Comment by jstedfast** [comment]
FWIW, this fix should be in 1.0.16 which I released this weekend to nuget.org
